packages/serverpod/lib/src/server/log_manager/log_manager.dart

```diff
@@ -316,7 +323,7 @@ class SessionLogManager {
     String? exception,
     StackTrace? stackTrace,
   }) async {
-    await _openStreamLogLock.synchronized(() {});
+    if (!_isLoggingOpened) await _openStreamLogLock.synchronized(() {});
```

Comment:
Question: For me this looks like we only wait on the lock if logging is not opened. But if the lock is never acquired we will just move past this line.

What is this guarding against?